src/d3a/models/area/redis_external_market_connection.py

```diff
@@ -104,47 +103,36 @@ def sub_to_external_channels(self) -> None:
 
     def set_grid_fees_callback(self, payload: Dict) -> Optional[Dict]:
         """Update the grid fees of the market."""
-        # TODO: This function should reuse the area_reconfigure_event function
-        # since they share the same functionality.
         if not (self._connected or self.is_aggregator_controlled):
             return None
         grid_fees_response_channel = f"{self.channel_prefix}/response/grid_fees"
         payload_data = (
             payload["data"] if isinstance(payload["data"], dict)
             else json.loads(payload["data"]))
         try:
-            validate_area(grid_fee_percentage=payload_data.get("fee_percent", None),
-                          grid_fee_constant=payload_data.get("fee_const", None))
+            self.area.area_reconfigure_event(
+                grid_fee_constant=payload_data.get("fee_percent", None),
+                grid_fee_percentage=payload_data.get("fee_const", None))
```

Comment:
You are setting the grid-fees before you have validated them.
The comment in line 107 was reffering to replace 
```python
self.area.grid_fee_percentage = payload_data["fee_percent"]
```
and
```python
self.area.grid_fee_constant = payload_data["fee_const"]
```
with 
```python
self.area.area_reconfigure_event(
                grid_fee_constant=payload_data.get("fee_percent", None),
                grid_fee_percentage=payload_data.get("fee_const", None))
```
The validation here and below should stay the sam (could be refactored of course), though.
BTW, this method can/should be unittested for exactly the validation (especially against the GlobalConfigFeeType).

Reply:
@hannesdiedrich I have already created dedicated unit tests to test the behavior upon submitting wrong GlobalConfigFeeType , you can check the `test_set_grid_fees_callback_conflicting_grid_fee_type_config` and `test_set_grid_fees_callback_valid_grid_fee_type_config`.

Furthermore, per your first point regarding the validation of the fees, the validation already exists in the area's [area_reconfigure_event](https://github.com/gridsingularity/gsy-e/blob/832a3799bd323dcb1abe5a783b234744b61dcb0c/src/gsy_e/models/area/__init__.py#L211) method, this is why I removed the redundant `validate_area` call from here.

Reply:
Darn, I was trapped inside one commit. Reviewed it now. Thanks a lot for the extensive tests!